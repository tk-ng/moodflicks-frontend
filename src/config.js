const PosterBaseUrl = "https://image.tmdb.org/t/p/";

export { PosterBaseUrl };
