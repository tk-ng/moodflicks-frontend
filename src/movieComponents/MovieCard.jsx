import {
	Card,
	CardBody,
	CardTitle,
	CardSubtitle,
	CardText,
	Button,
} from "reactstrap";
import { PosterBaseUrl } from "../config";
import { useNavigate } from "react-router";

export const MovieCard = ({
	id,
	overview,
	posterPath,
	releaseDate,
	title,
	editAuth,
	handleRemove,
}) => {
	const imageSrc = posterPath
		? PosterBaseUrl + "w185" + posterPath
		: "/img/placeholder.png";
	const navigate = useNavigate();

	return (
		<Card
			style={{
				width: "20rem",
			}}
			className="MovieCard m-1 p-3"
			onClick={() => navigate(`/movies/${id}`)}
		>
			<img alt="Movie Poster" src={imageSrc} />
			<CardBody>
				<CardTitle tag="h5">{title}</CardTitle>
				<CardSubtitle className="mb-2 text-light fs-6" tag="h6">
					released: {releaseDate}
				</CardSubtitle>
				<CardText className="MovieCardTruncate">{overview}</CardText>
				{editAuth && (
					<Button color="danger" onClick={(evt) => handleRemove(evt, id)}>
						Remove
					</Button>
				)}
			</CardBody>
		</Card>
	);
};
