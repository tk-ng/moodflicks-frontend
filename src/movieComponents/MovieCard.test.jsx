import { MovieCard } from "./MovieCard";
import { MemoryRouter } from "react-router";
import { render, screen } from "@testing-library/react";
import { PosterBaseUrl } from "../config";
import userEvent from "@testing-library/user-event";

describe("Render the MovieCard component", function () {
	const testMovie = {
		id: "testMovieId",
		overview: "testMovieOverview",
		posterPath: "testMoviePosterPath",
		releaseDate: "testMovieReleaseDate",
		title: "testMovieTitle",
		editAuth: false,
		handleRemove: vi.fn(),
	};
	it("mounts without crashing", async function () {
		const { container } = render(
			<MemoryRouter>
				<MovieCard />
			</MemoryRouter>
		);
		expect(container).toBeInTheDocument();
	});

	it("matches snapshot", async function () {
		const { asFragment } = render(
			<MemoryRouter>
				<MovieCard />
			</MemoryRouter>
		);
		expect(asFragment).toMatchSnapshot();
	});

	it("renders movie details WITHOUT the Remove buttons", async function () {
		render(
			<MemoryRouter>
				<MovieCard
					id={testMovie.id}
					overview={testMovie.overview}
					posterPath={testMovie.posterPath}
					releaseDate={testMovie.releaseDate}
					title={testMovie.title}
					editAuth={testMovie.editAuth}
					handleRemove={testMovie.handleRemove}
				/>
			</MemoryRouter>
		);
		expect(
			screen.getByText(testMovie.title, { selector: "h5" })
		).toBeInTheDocument();

		expect(
			screen.getByText(`released: ${testMovie.releaseDate}`, { selector: "h6" })
		).toBeInTheDocument();

		// The Remove button should not be displayed
		expect(
			screen.queryByText("Remove", { selector: "button" })
		).not.toBeInTheDocument();

		expect(screen.getByAltText("Movie Poster")).toHaveAttribute(
			"src",
			PosterBaseUrl + "w185" + testMovie.posterPath
		);
	});

	it("renders movie details WITH the Remove buttons", async function () {
		render(
			<MemoryRouter>
				<MovieCard
					id={testMovie.id}
					overview={testMovie.overview}
					posterPath={testMovie.posterPath}
					releaseDate={testMovie.releaseDate}
					title={testMovie.title}
					editAuth={true}
					handleRemove={testMovie.handleRemove}
				/>
			</MemoryRouter>
		);
		const user = userEvent.setup();

		expect(
			screen.getByText(testMovie.title, { selector: "h5" })
		).toBeInTheDocument();

		expect(
			screen.getByText(`released: ${testMovie.releaseDate}`, { selector: "h6" })
		).toBeInTheDocument();

		// The Remove button should be displayed
		const removeBtn = screen.getByText("Remove", { selector: "button" });
		expect(removeBtn).toBeInTheDocument();

		expect(screen.getByAltText("Movie Poster")).toHaveAttribute(
			"src",
			PosterBaseUrl + "w185" + testMovie.posterPath
		);

		await user.click(removeBtn);
		expect(testMovie.handleRemove).toHaveBeenCalledWith(
			expect.anything(),
			testMovie.id
		);
	});
});
