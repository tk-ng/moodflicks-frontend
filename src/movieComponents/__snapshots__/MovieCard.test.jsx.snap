// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Render the MovieCard component > matches snapshot 1`] = `[Function]`;

exports[`Render the MovieCard component > matches snapshot 2`] = `[Function]`;
